pages/embed/live/[embed_id].tsx

```diff
@@ -1,8 +1,51 @@
 import YstvHead from "../../../components/YstvHead";
-import Index from "../../../components/VideoPlayer";
+import VideoPlayer from "../../../components/VideoPlayer";
 
 import styles from "../embed.module.css";
+import { GetServerSideProps } from "next";
+import { channel } from "../../watch/live/[liveURLName]";
 
-export default function Embed({ video, time }) {
-  return <div className={styles.bg}>Embed for live video, coming soon</div>;
+export default function Embed({ channel }) {
+  const videoJSOptions = {
+    autoplay: false,
+    playbackRates: [0.5, 1, 1.25, 1.5, 2],
+    height: 500,
+    controls: true,
+    liveui: true,
+    fill: true,
+    sources: [
+      {
+        src: channel.outputURL,
+        type: "application/x-mpegURL",
+      },
+    ],
+  };
+
+  return (
+    <div className={styles.bg}>
+      <YstvHead title={`Watch Live - ${channel.name}`} />
+      <VideoPlayer options={videoJSOptions} time={0} />
+    </div>
+  );
 }
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  function redirect() {
+    context.res.statusCode = 302;
+    context.res.setHeader("Location", `/404`);
+  }
+
+  const channel: channel = await fetch(
+    `${process.env.REST_API}/v1/public/playout/channel/${context.query.embed_id}`
+  ).then((res) => {
+    if (!res.ok) {
+      redirect();
+    } else {
+      return res.json();
+    }
+  });
```

Comment:
I'm not too fussed at this stage, since I'm guessing the all API calls need updating as well. But it would be good to move all these channel get API calls to a separate API file and use that rather than fetch directly

Reply:
the duplication and mess is annoying me too don't worry (see #42), I'm hoping to fix this with a ystv npm library (maybe an auto generating one idk) that we can use for all the webapi dependant projects

Reply:
Yeah reasonable, wil need to get around to moving the my-tv one into a separate package at some point